### PR TITLE
did my own version of mergesort. Tested multiple times.

### PR DESCRIPTION
**Pull request by JonR2002**

### CODE REVIEW

**Comment by ndeanon25** [comment]
This merge sort is easy to follow. The comments are helpful.

File: dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java [hunk 1]
```diff
@@ -26,44 +29,68 @@ public static int[] insertionSort(int[] arr) {
...
-                
-            }
-            if (arr1[j] >= arr2[k]) {
-                newArray[i] = arr1[j];
+    
+    public static void mergeSortHelper(int[] arr, int start, int end) {
+        if (start < end) {
+            int mid = (start + end) / 2; // Calculate the midpoint of the subarray
+    
+            // Recursively sort the left and right halves
```
Comment (whereismyprozac) [inline dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java:41]: this probably works, good job

File: dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java [hunk 2]
```diff
@@ -26,44 +29,68 @@ public static int[] insertionSort(int[] arr) {
...
+                i++;
+            } else {
+                arr[k] = rightArray[j];
                 j++;
             }
-            else {
-                newArray[i] = arr2[k];
-                k++;
-            }
+            k++;
```
Comment (whereismyprozac) [inline dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java:79]: awesome.

**Review by whereismyprozac — COMMENTED** [review]
We've been trying to reach you concerning your vehicle's extended warranty. You should've received a notice in the mail about your car's extended warranty eligibility. Since we've not gotten a response, we're giving you a final courtesy call before we close out your file. Press 2 to be removed and placed on our do-not-call list. To speak to someone about possibly extending or reinstating your vehicle's warranty, press 1 to speak with a warranty specialist.

File: dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java [hunk 3]
```diff
@@ -26,44 +29,68 @@ public static int[] insertionSort(int[] arr) {
 
         return arr;
     }
-    public static int[] mergeSort(int[] array){
-        return mergeSortHelper(array, 0, array.length - 1);
-
```
Comment (Julianc02) [inline dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java:31]: Declares Merge sort helper method & iterates through the array

File: dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java (hunk 2 above)
Comment (Lucretian) [inline dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java:79]: This deals with the leftover indexes, adding them to the new array.

File: dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java [hunk 4]
```diff
@@ -26,44 +29,68 @@ public static int[] insertionSort(int[] arr) {
...
+        }
+    
+        while (j < rightLength) {
+            arr[k] = rightArray[j];
+            j++;
+            k++;
+        }
+    }
 }
+
```
Comment (Lucretian) [inline dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java:96]: This deals with the leftover indexes not added to the array.

File: dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java [hunk 5]
```diff
@@ -26,44 +29,68 @@ public static int[] insertionSort(int[] arr) {
...
+        int[] rightArray = new int[rightLength];
+    
+        // Copy data to temporary left and right arrays
+        for (int i = 0; i < leftLength; i++) {
+            leftArray[i] = arr[start + i];
+        }
+        for (int i = 0; i < rightLength; i++) {
+            rightArray[i] = arr[mid + 1 + i];
+        }
+
```
Comment (Julianc02) [inline dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java:67]: Arrays are divided into sub-arrays until they are individually sorted

**Pull request merged**